lib/screens/v2/disruption_diagram/builder.ex

```diff
@@ -0,0 +1,967 @@
+defmodule Screens.V2.DisruptionDiagram.Builder do
```

Comment:
Gave this file a second pair of eyes and I don't have anything to add. The iffy parts for me already had comments from Hannah. Any work on this in the future will definitely have a steep learning curve, but the logic is sound. Great work!